### PR TITLE
Task #224692 [BE] Prerak Status change and register audit log 

### PR DESCRIPTION
**Pull request by Tusharmahajan12**
https://tracker.tekdi.net/issues/224692
# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [ ] Code is formatted as per format decided
* [ ] Updated acceptance criteria in tracker
* [ ] Updated test cases in test-cases-tracker
* [ ] Updated new API endpoint if any in common postman collection
* [ ] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced user auditing capabilities for tracking user actions.
	- Added functionality to log detailed user action data, improving traceability and accountability.

- **Bug Fixes**
	- Changed the declaration of the `addAuditLogAction` method to improve type safety.

### CODE REVIEW

**Review by GitVivekHub — APPROVED** [review]

File: src/src/user/user.service.ts [hunk 1]
```diff
@@ -136,7 +136,22 @@ export class UserService {
...
 				data: data,
 			};
-
+			let auditData = {
+				userId: facilitator_user_id,
+				mw_userid: facilitator_user_id,
+				user_type: 'IP',
+				context: 'prerak.update.status',
+				context_id: facilitator_user_id,
+				subject: 'prerak',
```
Comment (sagarkoshti1990) [inline src/src/user/user.service.ts:145]: release prerak to facilitator

File: src/src/userauth/userauth.service.ts [hunk 2]
```diff
@@ -283,6 +283,24 @@ export class UserauthService {
 						acknowledgement_create_body,
 					);
 				}
+				let auditData = {
+					userId: result?.data?.id,
+					mw_userid: result?.data?.id,
+					user_type: 'Prerak',
+					context: 'prerak.created',
+					context_id: result?.data?.id,
+					subject: 'prerak',
```
Comment (sagarkoshti1990) [inline src/src/userauth/userauth.service.ts:292]: release prerak to facilitator

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

File: src/src/user/user.service.ts (hunk 1 above)
Comment (Tusharmahajan12) [reply src/src/user/user.service.ts]: ok

File: src/src/userauth/userauth.service.ts (hunk 2 above)
Comment (Tusharmahajan12) [reply src/src/userauth/userauth.service.ts]: done

**Pull request merged**